services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,264 @@
+import software.amazon.awssdk.services.cloudfront.internal.auth.Pem;
+import software.amazon.awssdk.services.cloudfront.internal.auth.Rsa;
+import software.amazon.awssdk.services.cloudfront.internal.cookie.DefaultCookiesForCannedPolicy;
+import software.amazon.awssdk.services.cloudfront.internal.cookie.DefaultCookiesForCustomPolicy;
+import software.amazon.awssdk.services.cloudfront.internal.url.DefaultSignedUrl;
+import software.amazon.awssdk.services.cloudfront.internal.utils.SigningUtils;
+import software.amazon.awssdk.services.cloudfront.model.CloudFrontSignerRequest;
+import software.amazon.awssdk.services.cloudfront.url.SignedUrl;
+import software.amazon.awssdk.utils.IoUtils;
+import software.amazon.awssdk.utils.StringUtils;
+
+/**
+ *
+ * Utilities for working with CloudFront distributions
+ * <p>
+ *     To securely serve private content by using CloudFront, you can require that users access your private content by using
+ *     special CloudFront signed URLs or signed cookies. You then develop your application either to create and distribute signed
+ *     URLs to authenticated users or to send Set-Cookie headers that set signed cookies for authenticated users.
+ * </p>
+ * <p>
+ *     Signed URLs take precedence over signed cookies. If you use both signed URLs and signed cookies to control access to the
+ *     same files and a viewer uses a signed URL to request a file, CloudFront determines whether to return the file to the
+ *     viewer based only on the signed URL.
+ * </p>
+ *
+ */
+@Immutable
+@ThreadSafe
+@SdkPublicApi
+public final class CloudFrontUtilities {
+
+    private static final String CLOUDFRONT_NET = "cloudfront.net";
+
+    private CloudFrontUtilities() {
+    }
+
+    /**
+     * Returns a signed URL with a canned policy that grants universal access to
+     * private content until a given date.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-canned-policy.html"
+     * >Creating a signed URL using a canned policy</a>.
+     *
+     * @param request
+     *            A CloudFrontSignerRequest configured with the following values:
+     *            resourceUrl, privateKey, keyPairId, expirationDate
+     * @return A signed URL that will permit access to a specific distribution
+     *         and S3 object.
+     */
+    public static SignedUrl getSignedUrlWithCannedPolicy(CloudFrontSignerRequest request) {
+        try {
+            String resourceUrl = request.resourceUrl();
+            String cannedPolicy = SigningUtils.buildCannedPolicy(resourceUrl, request.expirationDate());
+            byte[] signatureBytes = SigningUtils.signWithSha1Rsa(cannedPolicy.getBytes(UTF_8), request.privateKey());
+            String urlSafeSignature = SigningUtils.makeBytesUrlSafe(signatureBytes);
+            String protocol = resourceUrl.substring(0, resourceUrl.indexOf("://"));
```

Comment:
Can we just do URI.create(resourceUrl) to extract the protocol, domain and path? That feels safer than trying to parse the URL ourselves.

Reply:
agreed, updated to using URI